### PR TITLE
[RISCV] Rework IDiv and FDiv pipes on SiFive7

### PR DESCRIPTION
**Pull request by michaelmaitland**
Set BufferSize=0 and remove Super pipes for these resources.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVSchedSiFive7.td [hunk 1]
```diff
@@ -213,12 +213,12 @@ let SchedModel = SiFive7Model in {
 let BufferSize = 0 in {
 def SiFive7PipeA       : ProcResource<1>;
 def SiFive7PipeB       : ProcResource<1>;
+def SiFive7IDiv        : ProcResource<1> { let Super = SiFive7PipeB; } // Int Division
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVSchedSiFive7.td:216]: I don't know why these need a Super class. Their usages already say SiFive7PipeB in the ResourceCycles explicitly.

**Review by wangpc-pp — COMMENTED** [review]
The PR description says `Super` is removed but we still have it?

File: llvm/lib/Target/RISCV/RISCVSchedSiFive7.td (hunk 1 above)
Comment (michaelmaitland) [reply llvm/lib/Target/RISCV/RISCVSchedSiFive7.td:216]: I didn't mean to keep the Super class. As @wangpc-pp points out that the description of the PR says it is removed. I will push a fixup.

**Review by wangpc-pp — APPROVED** [review]
LGTM.
(Is this not covered by some llvm-mca tests?)

**Review by wangpc-pp — APPROVED** [review]

**Pull request merged**